contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,295 @@
+
+By publishing core metrics, the summary API is relieved of its responsibility to provide metrics to system components.  This will allow the summary API to evolve into a much richer set of metrics for monitoring, since the overhead burden falls on the third party monitoring pipeline.
+
+cAdvisor is structured to collect metrics on an interval, which is appropriate for a stand-alone metrics collector.  However, many functions in the kubelet are latency-sensitive (eviction, for example), and would benifit from a more "On-Demand" metrics collection design.
+
+### Proposal
+This proposal is to use this set of core metrics, collected by the kubelet, and used solely by kubernetes system components to support resource feasibility checking and resource management on the node.
+
+The target "Users" of this set of metrics are kubernetes components.  This set of metrics itself is not designed to be user-facing, but is designed to be general enough to support user-facing components.
+
+### Non Goals
+Everything covered in the [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) design doc will not be covered in this proposal.  This includes the third party metrics pipeline, and the methods by which the metrics found in this proposal are provided to other kubernetes components.
+
+Integration with CRI will not be covered in this proposal.  In future proposals, integrating with CRI may provide a better abstraction of information required by the core metrics pipeline to collect metrics.
+
+## Design
+This design covers only the Core Metrics Pipeline.
+
+High level requirements for the design are as follows:
+ - Do not break existing users.  We should continue to provide the full summary API as an optional add-on.  Once the monitoring pipeline is completed, the summary API will be provided by the monitoring pipeline, possibly through a stand-alone version of cAdvisor.
+ - The kubelet collects the minimum possible number of metrics for complete portable kubernetes functionalities.
+ - Metrics can be fetched "On Demand", giving the kubelet more up-to-date stats.
+
+More details on how these high level goals will be achieved can be found in the Implementation Plan.
+
+This Core Metrics API will be versioned to account for version-skew between kubernetes components.
+
+This proposal purposefully omits many metrics that may eventually become core metrics.  This is by design.  Once metrics are needed to support resource feasibility checking or node resource management, they can be added to the core metrics API.
+
+### Metric Requirements
+The core metrics api is designed to provide metrics for two use-cases within kubernetes:
+ - Resource Feasibility Checking
+ - Node Resource Management
+
+Many kubernetes system components currently support these features.  Many more components that support these features are in development.
```

Comment:
Refers to the above list of "use-cases"  changed the above to explicitly call them features.